.pre-commit-config.yaml

```diff
@@ -159,6 +159,12 @@ repos:
         types_or: [python, pyi]
         args: [--project=server/]
 
+      - id: deptry
+        name: deptry
+        entry: poetry --directory ./server run python -c 'import os;os.chdir("./server");os.execlp("deptry", "deptry", ".");'
```

Comment:
Can't we just do 

```shell
poetry --directory ./server run deptry
```

? or it's the same reason as for `ruff` ?

Reply:
Yes, it was a similar issue: `extend_exclude` and other default patterns did not work. 

However, now that you mention it, I manage to get it work by just prefixing "server" in the `extend_exclude` pattern in `pyproject.toml`:
```toml
[tool.deptry]
extend_exclude = [
    "server/tests/",
    "server/parsec/cli/testbed.py",
]
```
And then this works in `.pre-commit-config.yaml`:
```yaml
        entry: poetry --directory ./server run deptry server
        pass_filenames: false  # deptry does not expect filenames, only input dir for scan 
        args: [--config=server/pyproject.toml]  # otherwise deptry does not find it 
```
It seems better, but I don't know if its robust enough, WDYT?

Reply:
I prefer what you've just suggested, that provide a command that I can copy/paste (that is simpler to understand).

> You could flatten it a little more if you add the arguments in `args` in `entry`

Reply:
[`1d438be` (#7323)](https://github.com/Scille/parsec-cloud/pull/7323/commits/1d438be6fe60ae6ecbaa1d3a889a6295a0d37a1b)